pycrostates/segmentation/entropy.py

```diff
@@ -0,0 +1,584 @@
+    %(ignore_self)s
+    %(log_base)s
+
+    Returns
+    -------
+    h : float
+        The Shannon entropy of the sequence.
+    """
+    _check_type(labels, (np.ndarray,), "labels")
+    if state_to_ignore is not None:
+        _ensure_int(state_to_ignore, "state_to_ignore")
+    log_base = _check_log_base(log_base)
+
+    h = _joint_entropy_history(
+        labels, k=1, state_to_ignore=state_to_ignore, log_base=log_base
+    )
+    return h
+
+
+@fill_doc
+def entropy(
+    segmentation: [Segmentation, NDArray[int]],
+    state_to_ignore: Optional[int] = -1,
+    ignore_self: bool = False,
+    log_base: Union[float, str] = 2,
+):
+    r"""Compute the Shannon entropy of a symbolic sequence.
+
+    Compute the Shannon entropy\ :footcite:p:`shannon1948mathematical` of the microstate
+    symbolic sequence.
+
+    Parameters
+    ----------
+    %(segmentation_or_labels)s
+    %(state_to_ignore)s
+    %(ignore_self)s
+    %(log_base)s
+
+    Returns
+    -------
+    h : float
+        The Shannon entropy of the sequence.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    labels = _check_segmentation(segmentation)
+    if state_to_ignore is not None:
+        state_to_ignore = _ensure_int(state_to_ignore, "state_to_ignore")
+    _check_type(ignore_self, (bool,), "ignore_self")
+    log_base = _check_log_base(log_base)
+    # ignore transition to itself (i.e. AAABBBBC -> ABC)
+    if ignore_self:
+        labels = np.array([s for s, _ in itertools.groupby(labels)])
+
+    h = _entropy(labels, state_to_ignore=state_to_ignore, log_base=log_base)
+    return h
+
+
+# -- excess entropy rate ---------------------------------------------------------------
+@fill_doc
+def _excess_entropy_rate(
+    labels: NDArray[int],
+    history_length: int,
+    state_to_ignore: Optional[int] = -1,
+    log_base: Union[float, str] = 2,
+    n_jobs: int = 1,
+):
+    """Estimate the entropy rate and the excess_entropy from a linear fit.
+
+    Parameters
+    ----------
+    %(labels_info)s
+    history_length : int
+        Maximum history length in sample to estimate the excess entropy rate.
+    %(state_to_ignore)s
+    %(log_base)s
+    %(n_jobs)s
+
+    Returns
+    -------
+    a: float
+        Entropy rate (slope).
+    b: float
+        Excess entropy (intercept).
+    residual: float
+        Sum of squared residuals of the least squares fit.
+    lags: array of shape (history_length,)
+        Lag values in sample used for the fit.
+    joint_entropies: array of shape (history_length,)
+        Joint entropy value for each lag.
+    """
+    lags = np.arange(1, history_length + 1)
+    parallel, p_fun, _ = parallel_func(_joint_entropy_history, n_jobs, total=len(lags))
+    runs = parallel(
+        p_fun(labels, k, state_to_ignore=state_to_ignore, log_base=log_base)
+        for k in lags
+    )
+
+    (a, b), residuals, _, _, _ = np.polyfit(lags, runs, 1, full=True)
+    return (a, b, residuals[0], lags, runs)
```

Comment:
Function called only once in `excess_entropy_rate`. Any particular reason to split those 5 lines in a separate function? Else I would prefer to keep the code short and to move those 5 lines below in the `excess_entropy_rate` function.